### PR TITLE
Cardano data bridge

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

File: spectrum-cardano-connector/resources/serialized_cardano_blocks
Comment (kettlebell) [inline spectrum-cardano-connector/resources/serialized_cardano_blocks:1]: Serialized bytes of 30 blocks from Cardano testnet.

File: spectrum-chain-connector/src/lib.rs
Comment (kettlebell) [inline spectrum-chain-connector/src/lib.rs:1]: Code here is moved from `spectrum-ergo-connector`, as it's useful for the Cardano bridge too.

File: spectrum-cardano-connector/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+mod rocksdb;
+
+pub struct CardanoDataBridge {
+    pub receiver: tokio::sync::mpsc::Receiver<TxEvent<Vec<u8>>>,
+    tx_start: tokio::sync::oneshot::Sender<()>,
+}
+
+impl DataBridge for CardanoDataBridge {
+    type TxType = Vec<u8>;
```
Comment (kettlebell) [inline spectrum-cardano-connector/src/lib.rs:20]: Have to deal with serialized bytes of the transactions due to Pallas' use of copy-on-write (Cow) primitives. See: https://github.com/txpipe/pallas/blob/ea9c4e8b59fe5cbaaf73c649b44347a066f70960/pallas-traverse/src/lib.rs#L78

File: spectrum-cardano-connector/src/rocksdb.rs [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+        .await
+    }
+}
+
+#[derive(Clone)]
+pub struct Block {
+    pub id: Hash<32>,
+    pub parent_id: Hash<32>,
+    pub slot: u64,
+    pub block_number: u64,
```
Comment (kettlebell) [inline spectrum-cardano-connector/src/rocksdb.rs:324]: Can't use slot number to measure maximum rollback length. The block number here is used, but I'm not sure exactly what it represents (I searched the cardano docs but no luck). It's definitely not height from genesis block though.

**Review by oskin1 — APPROVED** [review]

**Pull request merged**